tests/agent/test_event_handler.py

```diff
@@ -20,76 +46,152 @@
 pytestmark = pytest.mark.unit
 
 
-def test_event_handler_raises_for_unknown_event(mock_context):
-    event = UnknownEvent()
-    correlation_id = "74842258-803a-48ca-8921-eaf2802c14e2"
-    handler = EventHandler(context=mock_context)
+class TestEventHandler:
+    def test_event_handler_raises_for_unknown_event(self, mock_context):
+        event = UnknownEvent()
+        correlation_id = "74842258-803a-48ca-8921-eaf2802c14e2"
+        handler = EventHandler(context=mock_context)
+
+        with pytest.raises(UnknownEventError):
+            handler.handle_event(event=event, id=correlation_id)
+
+    @pytest.mark.parametrize(
+        "event_name, event, action_type",
+        [
+            (
+                "RunMissingnessDataAssistantEvent",
+                RunMissingnessDataAssistantEvent(
+                    datasource_name="test-datasource",
+                    data_asset_name="test-data-asset",
+                ),
+                RunMissingnessDataAssistantAction,
+            ),
+            (
+                "RunOnboardingDataAssistantEvent",
+                RunOnboardingDataAssistantEvent(
+                    datasource_name="test-datasource", data_asset_name="test-data-asset"
+                ),
+                RunOnboardingDataAssistantAction,
+            ),
+            (
+                "RunCheckpointEvent",
+                RunCheckpointEvent(
+                    checkpoint_id="3ecd140b-1dd5-41f4-bdb1-c8009d4f1940",
+                    datasource_names_to_asset_names={"Data Source name": {"Asset name"}},
+                ),
+                RunCheckpointAction,
+            ),
+            (
+                "DraftDatasourceConfigEvent",
+                DraftDatasourceConfigEvent(config_id=uuid4()),
+                DraftDatasourceConfigAction,
+            ),
+            (
+                "ListTableNamesEvent",
+                ListTableNamesEvent(datasource_name="test-datasource"),
+                ListTableNamesAction,
+            ),
+            (
+                "RunColumnDescriptiveMetricsEvent",
+                RunColumnDescriptiveMetricsEvent(
+                    datasource_name="test-datasource", data_asset_name="test-data-asset"
+                ),
+                ColumnDescriptiveMetricsAction,
+            ),
+        ],
+    )
+    def test_event_handler_handles_all_events(
+        self,
+        mock_context,
+        mocker: MockerFixture,
+        event_name: str,
+        event: Event,
+        action_type: type[AgentAction[Any]],
+    ):
+        action = mocker.MagicMock(autospec=action_type)
+        mocker.patch("great_expectations_cloud.agent.event_handler._GX_MAJOR_VERSION", "1")
+
+        mocker.patch.dict(_EVENT_ACTION_MAP, {"1": {event_name: action}})
```

Comment:
From synchronous review: this change triggered the parameterization of the test.